lib/chef/knife/supermarket_list.rb

```diff
@@ -37,23 +37,34 @@ class SupermarketList < Knife
         default: "https://supermarket.chef.io",
         proc: Proc.new { |supermarket| Chef::Config[:knife][:supermarket_site] = supermarket }
 
+      option :sort_by,
+        long: "--sort-by SORT",
+        description: "Use to sort the records",
+        in: %w{recently_updated recently_added most_downloaded most_followed}
+
+      option :owned_by,
+        short: "-o USER",
+        long: "--owned-by USER",
+        description: "Show cookbooks that are owned by the USER"
+
       def run
         if config[:with_uri]
-          cookbooks = {}
-          get_cookbook_list.each { |k, v| cookbooks[k] = v["cookbook"] }
-          ui.output(format_for_display(cookbooks))
+          ui.output(format_for_display(get_cookbook_list))
         else
-          ui.msg(ui.list(get_cookbook_list.keys.sort, :columns_down))
+          ui.msg(ui.list(get_cookbook_list.keys, :columns_down))
         end
       end
 
-      def get_cookbook_list(items = 10, start = 0, cookbook_collection = {})
+      def get_cookbook_list(items = 1000, start = 0, cookbook_collection = {})
```

Comment:
Let's just set this to 9999999 and avoid the whole thing. You probably want to add a comment that we're using 9999999 to avoid pagination entirely.